tools/shell/tests/conftest.py

```diff
@@ -0,0 +1,175 @@
+        self.arguments.append(arg)
+        return self
+
+    def statement(self, stmt):
+        self.statements.append(stmt)
+        return self
+
+    def add_arguments(self, args: List):
+        self.arguments.extend(args)
+        return self
+
+    def input_file(self, file_path):
+        self.input = file_path
+        return self
+
+    def output_file(self, file_path):
+        self.output = file_path
+        return self
+
+    # Test Running methods
+
+    def get_command(self, cmd: str) -> List[str]:
+        command = self.arguments
+        if self.input:
+            command += [cmd]
+        return command
+
+    def get_input_data(self, cmd: str):
+        if self.input:
+            input_data = open(self.input, 'rb').read()
+        else:
+            input_data = bytearray(cmd, 'utf8')
+        return input_data
+
+    def get_output_pipe(self):
+        output_pipe = subprocess.PIPE
+        if self.output:
+            output_pipe = open(self.output, 'w+')
+        return output_pipe
+
+    def get_statements(self):
+        result = ""
+        statements = []
+        for statement in self.statements:
+            if statement.startswith('.'):
+                statements.append(statement)
+            else:
+                statements.append(statement + ';')
+        return '\n'.join(statements)
+
+    def get_output_data(self, res):
+        if self.output:
+            stdout = open(self.output, 'r').read()
+        else:
+            stdout = res.stdout.decode('utf8').strip()
+        stderr = res.stderr.decode('utf8').strip()
+        return stdout, stderr
+
+    def run(self):
+        statements = self.get_statements()
+        command = self.get_command(statements)
+        input_data = self.get_input_data(statements)
+        output_pipe = self.get_output_pipe()
+
+        res = subprocess.run(command, input=input_data, stdout=output_pipe, stderr=subprocess.PIPE)
+
+        stdout, stderr = self.get_output_data(res)
+        return stdout, stderr, res.returncode
+
+
+@pytest.fixture()
+def shell(request):
+    custom_arg = request.config.getoption("--shell-binary")
+    if not custom_arg:
+        raise ValueError("Please provide a shell binary path to the tester, using '--shell-binary <path_to_cli>'")
+    return custom_arg
+
+
+@pytest.fixture()
+def random_filepath(request, tmp_path):
+    tmp_file = tmp_path / "random_import_file"
+    return tmp_file
+
+
+@pytest.fixture()
+def generated_file(request, random_filepath):
+    param = request.param
+    tmp_file = random_filepath
+    with open(tmp_file, 'w+') as f:
+        f.write(param)
+    return tmp_file
+
+
+def assert_expected_res(out, expected, status, err):
+    if isinstance(expected, list):
+        expected = '\n'.join(expected)
+    if expected not in out:
+        print("Exit code:", status)
+        print("Captured stderr:", err)
+        print("Actual result:", out)
+        assert False
+    assert status == 0
+    assert True
```

Comment:
This is a no-op

Reply:
Not sure what you mean by that, just wanted to mark "passed" for every test that doesnt fail

But I guess passed assertions are ignored?